montblanc/impl/rime/tensorflow/rime_ops/zernike_op_gpu.cuh

```diff
@@ -149,33 +164,61 @@ __global__ void zernike_dde_zernike(
     int ant = blockIdx.y * blockDim.y + threadIdx.y;
     int time = blockIdx.z * blockDim.z + threadIdx.z;
 
-  int i = threadIdx.z * 4 + (threadIdx.x & 0x03);
+    if(corr >= 4 || chan >= nchan || ant >= na || time >= ntime) return;
```

Comment:
Use a constexpr outside the kernel (in fact, in the zernike_dde.h) file and use it throughout the code, rather than relying on hard-coded constants.

```cpp
constexpr int _ZERNIKE_CORRS = 4
```